@xen-orchestra/lite/src/components/Tabs.tsx

```diff
@@ -0,0 +1,84 @@
+import Box from '@mui/material/Box'
+import React from 'react'
+import Tab from '@mui/material/Tab'
+import TabContext from '@mui/lab/TabContext'
+import TabList from '@mui/lab/TabList'
+import TabPanel from '@mui/lab/TabPanel'
+import { RouteComponentProps } from 'react-router-dom'
+import { withState } from 'reaclette'
+import { withRouter } from 'react-router'
+
+interface ParentState {}
+
+interface State {
+  pathname: string
+}
+
+interface TabType {
+  component?: React.ReactNode
+  disabled: boolean
+  label: React.ReactNode
+  pathname: string
+}
+
+interface Props {
+  history: RouteComponentProps['history']
```

Comment:
You can avoid to rewrite `history` by using RouteComponentProps interface.
By inheritance of this interface, you ensure to forward to `typescript` the "props type" needed for the component and make it easier to reuse.

```suggestion
interface Props extends RouteComponentProps {
```

Reply:
In order not to raise the question of what we are extending, I added explicitly the history props.

Reply:
But use `withRouter` without extends this interface generate some `typescript` errors no ?

Reply:
I found that weird to don't use the inerithance here.
It avoid to rewirte all props given by `withRouter` and avoid some`typescript` errors.
If is not clear, maybe a comment can be usefull.
Something like that: `RouteComponentProps ensure the compatiblity with 'withRouter'` or link to this [doc](https://www.reactandtypescript.dev/examples/react-router#withrouter) ?